streams/shared/src/main/scala/zio/stream/ZChannel.scala

```diff
@@ -690,12 +690,14 @@ sealed trait ZChannel[-Env, -InErr, -InElem, -InDone, +OutErr, +OutElem, +OutDon
                            permits
                              .withPermit(
                                latch.succeedUnit *>
-                                 f(outElem)
-                                   .catchAllCause(cause =>
-                                     failureRef.update(_ && cause).unless(cause.isInterruptedOnly) *>
-                                       errorSignal.succeedUnit *>
+                                 f(outElem).catchAllCause { cause =>
+                                   @inline def continue =
```

Comment:
AFAICT can make this into a `val` right? `@inline` has no effect in Scala 3 so probably best to make it into a val

Reply:
Done!